src/gen_experiments/pdes.py

```diff
@@ -0,0 +1,235 @@
+        "initial_condition": np.reshape(np.zeros((10, 10)), (100,)),
+        "spatial_args": [0.01, 10],
+        "time_args": [0.01, 10]
+    },
+    "burgers3D": {
+        "rhsfunc": {
+            "func": burgers3D,
+            "dimension": 3
+        },
+        "input_features": ["u", "ux", "uxx", "uy", "uyy", "uz", "uzz"],
+        "initial_condition": np.reshape(np.zeros((10, 10, 10)), (1000,)),
+        "spatial_args": [0.01, 10],
+        "time_args": [0.01, 10]
+    },
+    "kuramoto-sivashinsky": {
+        "rhsfunc": {
+            "func": ks,
+            "dimension": 1
+        },
+        "input_features": ["u", "ux", "uxx", "uxxxx"],
+        "initial_condition": np.zeros(100),
+        "spatial_args": [0.01, 100],
+        "time_args": [0.01, 100]
+    }
+}
+
+def run_pde(
+    seed: float,
+    /,
+    group: str,
+    # sim_params: dict,
+    diff_params: dict,
+    feat_params: dict,
+    opt_params: dict,
+    display: bool = True,
+    return_all: bool = True,
+) -> dict:
+    rhsfunc = pde_setup[group]["rhsfunc"]["func"]
+    input_features = pde_setup[group]["input_features"]
+    initial_condition = pde_setup[group]["initial_condition"]
+    spatial_args = pde_setup[group]["spatial_args"]
+    time_args = pde_setup[group]["time_args"]
+    dimension = pde_setup[group]["rhsfunc"]["dimension"]
+    coeff_true = pde_setup[group]["coeff_true"]
+    spatial_grid = pde_setup[group]["spatial_grid"]
+    try:
+        time_args = pde_setup[group]["time_args"]
+    except KeyError:
+        time_args = [0.01, 10]
+    nonnegative = False
+    dt, t_train, x_train, x_test, x_dot_test, x_train_true = gen_pde_data(
+        rhsfunc,
+        initial_condition,
+        spatial_args,
+        dimension,
+        seed,
+        noise_abs=0,
+        dt=time_args[0],
+        t_end=time_args[1]
+    )
+    model = _make_model(input_features, dt, diff_params, feat_params, opt_params)
+
+    model.fit(x_train[0], quiet=True)
+    coeff_true, coefficients, feature_names = unionize_coeff_matrices(model, coeff_true)
+
+    # make the plots
+    if display:
+        model.print()
+        compare_coefficient_plots(
+            coefficients,
+            coeff_true,
+            input_features=input_features,
+            feature_names=feature_names,
+        )
+        smoothed_last_train = model.differentiation_method.smoothed_x_
+        plot_pde_training_data(x_train[-1], x_train_true, smoothed_last_train)
+
+    # calculate metrics
+    metrics = coeff_metrics(coefficients, coeff_true)
+    metrics.update(integration_metrics(model, x_test, t_train, x_dot_test))
+    if return_all:
+        return (
+            metrics, {
+                "t_train": t_train,
+                "x_train": x_train,
+                "x_test": x_test,
+                "x_dot_test": x_dot_test,
+                "x_train_true": x_train_true,
+                "model": model,
+            }
+        )
+    return metrics
```

Comment:
EOF new line

Reply:
still relevant

Reply:
NVM I inspected the hexdump - this is just how github renders files.